send-nft/index.js

```diff
@@ -108,36 +104,65 @@ async function run() {
     const signingClient = await createNFTSigningClient(signer);
     const client = signingClient.getSigningStargateClient();
 
+    const hasCsvMemo = !!data.find((e) => !!e.memo);
+
     const msgAnyArray = [];
-    data.forEach((e) => {
+    for (let i = 0; i < data.length; i += 1) {
+      const e = data[i];
       const removed = nftsDataObject[e.classId].splice(0, 1);
-      const msgAny = formatMsgSend(
-        firstAccount.address,
-        e.address,
-        e.classId,
-        removed[0].id,
-      );
-      msgAnyArray.push(msgAny);
-    });
+      if (hasCsvMemo) {
+        const tx = await client.sign(
+          firstAccount.address,
+          msgAnyArray,
+          {
+            amount: [
+              {
+                denom: DENOM,
+                amount: `${new BigNumber(1).shiftedBy(amountNeedDigits).toFixed(0)}`,
+              },
+            ],
+            gas: `${new BigNumber(1).shiftedBy(gasNeedDigits).toFixed(0)}`,
+          },
+          MEMO,
+        );
+        try {
+          await client.broadcastTx(tx, 1000, 1000);
```

Comment:
Do we expect to sign tx separately if any of the rows in .csv has the memo field?

Reply:
we would need 1 tx per memo, so yes

Reply:
Oh, I see

Reply:
Oh, I see. 
BTW, do we prefer to push those messages that don't have a memo to the array and sign them at once, or it's OK to keep them signed and broadcasted one by one?

Reply:
signing doesnt takes much time except fetching sequence, so they should be the same
on second thought, you reminded me that i should increment and assign sequence manually ...